evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py

```diff
@@ -50,8 +50,11 @@ def __init__(self, boosting_type="gbdt", learning_rate=0.1, n_estimators=100, ma
         # when boosting type is random forest (rf), LightGBM requires bagging_freq == 1 and  0 < bagging_fraction < 1.0
         if boosting_type == "rf":
             lg_parameters['bagging_freq'] = 1
+        # when boosting type is goss, LightGBM requires bagging_fraction == 1
+        elif boosting_type == "goss":
+            lg_parameters['bagging_fraction'] = 1
         # avoid lightgbm warnings having to do with parameter aliases
-        if lg_parameters['bagging_freq']:
+        if lg_parameters['bagging_freq'] is not None:
```

Comment:
@angela97lin I think the `if/elif` sections are fine, since for `rf`,  we need to do bagging, whereas for `goss`, lightgbm doesn't accept bagging. For the last check though, I would set `subsample` and `subsample_freq` to None by default since as long as bagging values are passed in, it'll throw the warning if subsample values aren't None.

Reply:
@bchen1116 Just to make sure I'm understanding, are you suggesting that rather than checking if `bagging_freq` is None, _always_ update `subsample` and `subsample_freq` to `None`? :o

Reply:
@dsherry @bchen1116 Oooo. Based on what you guys have said, does that mean I should be checking if either `bagging_freq` or `bagging_fraction` are set to None? (currently just if `lg_parameters['bagging_freq'] is not None:`)

Reply:
@angela97lin yep I think so!

Perhaps the quickest way to resolve this is to write a unit test where we try various combos of all these params, run `LightGBMClassifier.fit` on a tiny dataset each time, and assert there are no warnings in `warnings` or in the output?

Reply:
Right that should do it. I think if `bagging_freq` is None, then it should ignore `bagging_fraction`, but I do think the safest way to do it would be to just set `subsample/subsample_freq` to none as long as either of the bagging args are set. Sorry, didn't see this earlier!